src/main/java/levi9/UserService/service/impl/UserServiceImpl.java

```diff
@@ -59,4 +60,21 @@ public UserDto loadUserByUsername(String username) {
     }
 
 
+
+    @Override
+    public UserDto editUserDetails(String username,UserDto userDetails ) {
+            User userDetails2 = mapper.map(userDetails,User.class);
```

Comment:
No need for this object, properties can be copied from `userDetails` directly onto `oldUser`